lib/inspec/formatters/base.rb

```diff
@@ -84,6 +85,60 @@ def results
 
     private
 
+    def all_unique_controls
+      unique_controls = Set.new
+      run_data[:profiles].each do |profile|
+        profile[:controls].map { |control| unique_controls.add(control) }
+      end
+
+      unique_controls
+    end
+
+    def statistics
```

Comment:
I am a little bit worried of putting the stats into the reporter. Because this leads to additional problems/questions: stats per profile etc.

Reply:
Yeah, we need it here so the base runner knows what exit code to use. If we do it upstream then it would be hard to capture an accurate exit code for a rspec format. Removing the major/minor/critical will minimize this though.